### ISSUE TITLE
Generating a histogram for an array of identical values less then 1 throws a Math::DomainError

### ISSUE THREAD

**Issue by kurttomlinson** [jtprince/histogram#8]
How to reproduce:
```
require 'histogram/array'
[1e-1, 1e-1, 1e-1].histogram
```

Error and stack trace:
```
Math::DomainError: Numerical argument is out of domain - "sqrt"
        from C:/Ruby21/lib/ruby/gems/2.1.0/gems/histogram-0.2.2.0/lib/histogram.rb:43:in `sqrt'
        from C:/Ruby21/lib/ruby/gems/2.1.0/gems/histogram-0.2.2.0/lib/histogram.rb:43:in `sample_stats'
        from C:/Ruby21/lib/ruby/gems/2.1.0/gems/histogram-0.2.2.0/lib/histogram.rb:97:in `number_of_bins'
        from C:/Ruby21/lib/ruby/gems/2.1.0/gems/histogram-0.2.2.0/lib/histogram.rb:208:in `histogram'
        from (irb):2
        from C:/Ruby21/bin/irb:11:in `<main>'
```

**Comment by jtprince** [comment]
thanks for the report.  I'll look into it, but also feel free to submit a pull-request if you patch it sooner than me.

**Comment by jtprince** [comment]
I patched this up, and the patch also deals with a few other edge cases that were also not handled well before.  Check for release in next few minutes.

**Comment by jtprince** [comment]
released histogram 0.2.3.0 to rubygems.org

**Comment by kurttomlinson** [comment]
Thanks!
